Solutions/Corvus.SpecFlow.Extensions/Corvus/SpecFlow/Extensions/FunctionsController.cs

```diff
@@ -143,7 +149,7 @@ public async Task StartFunctionsInstance(FeatureContext featureContext, Scenario
         /// </summary>
         public void TeardownFunctions()
         {
-            List<Exception> aggregate = null;
```

Comment:
Funnily enough, that's exactly what I did to start with.

But the intent of the code is that if 0 exceptions have been captured, the aggregate exception is not thrown. So to me, looking at the count rather than doing a null check is more in line with that. Yes, the ultimate effect is the same but I think it makes the code communicate it's intent better. A quick scan of the code now makes it obvious what the condition for throwing the AggregateException is, whereas previously there'd be the extra step of understanding the cicumstance under which the list was null.

Thoughts?

Reply:
I find it mildly annoying that we create a list that might never be used, but I can see the logic in your argument. And it's not like this is a hyper-GC-sensitive thing... So you've convinced me.